lib/zhr_devs/bakery_integration/commands/gen_multiplayer.ex

```diff
@@ -51,12 +58,26 @@ defmodule ZhrDevs.BakeryIntegration.Commands.GenMultiplayer do
     end)
   end
 
-  def on_success(output_file_path) do
+  def on_success(opts) do
+    output_file_path = Keyword.fetch!(opts, :output_json_path)
+    task_uuid = Keyword.fetch!(opts, :task_uuid)
+    solution_uuid = Keyword.fetch!(opts, :solution_uuid)
+
     if File.exists?(output_file_path) do
       Logger.info("Successfully generated tournament output: #{inspect(output_file_path)}")
 
-      # We want to issue a command here for sure.
-      :ok
+      complete_check_solution = %CompleteCheckSolution{
+        solution_uuid: solution_uuid,
+        task_uuid: task_uuid,
+        score: extract_score!(output_file_path)
+      }
+
+      :ok = persist_output(output_file_path, task_uuid, solution_uuid)
+
+      # We now want to safely delete the original output.json (maybe with the whole tournament dir?)
```

Comment:
I think bakery handles this, does rerunning work during your tests?

Reply:
Yes, it working when I submit multiple solutions